### ISSUE TITLE
DoH requests timing out

### ISSUE THREAD

**Issue by SnwMds** [bromite/bromite#437]
### Bromite version

Version: `78.0.3904.130`  
Arch: `arm`  
Android version: `8.1`  
Device model: `AllCall_S1_X`

### Is this bug about the SystemWebView?

No

### Is the bug reproducible with latest version?

Yes

### Can the bug be reproduced with corresponding Chromium version?

No, Chromium doesn't have this feature

### Is the bug a crash?

No

### Describe the bug

Sometimes Bromite can't resolve domain names. When trying to access some websites, it displays the error `ERR_DNS_TIMED_OUT`. This does not happens when DNS-over-HTTPS is disabled.

### Steps to reproduce the bug

1. Go to 'Settings > Privacy > Secure DNS'
2. Enter any valid DoH template
3. Save it and enable the Secure DNS
4. Browse some websites
5. See the error

### Expected behavior

Bromite can resolve the names of the domains accessed, enabling the user to access the desired content.

### Screenshots

See [screenshot-1.jpg](http://snwmds.org/images/screenshot-1.jpg), [screenshot-2.jpg](http://snwmds.org/images/screenshot-2.jpg) and [screenshot-3.jpg](http://snwmds.org/images/screenshot-3.jpg)

**Comment by sviluppoSoftware** [comment]
have same problem sometime, unfortunately there doesn't seem to be a precise cause so it is difficult to reproduce it, however it happened both on wifi and mobile data and i have try different DOH dns server...

**Comment by csagan5** [comment]
This happens when the network is slow; I am going to check what timeout values are used but there is nothing to be done if the timeout being used is already good enough as upstream sets it.

**Comment by SnwMds** [comment]
@csagan5, the problem is not the speed of the internet connection. The error is displayed right after trying to access the content. This means that there is no timeout between the first request (user action) and the DNS resolver response (Bromite request).

**Comment by csagan5** [comment]
> This means that there is no timeout between the first request

A 10ms timeout would produce the same effect.

**Comment by csagan5** [comment]
@SnwMds @sviluppoSoftware can you please test with latest version `79.0.3945.100`? There are no functional changes but if you grep the logcat like this:
```
adb logcat | grep DoH
```

You can report back here the timeout being used and hopefully we can further troubleshoot the issue.

**Comment by SnwMds** [comment]
@csagan5, sorry for the delay in replying. I tested version `79.0.3945.100` with DNS-over-HTTPS enabled. These are the logs I got:  

`12-30 23:54:27.644530  9508  9508 I chromium: [INFO:system_network_context_manager.cc(145)] DoH templates: 'https://doh-fi.blahdns.com/dns-query' with mode secure
12-30 23:54:27.669293  9508  9568 I chromium: [INFO:dns_session.cc(331)] DoH: using timeout: 1.01 s`

**Comment by csagan5** [comment]
@SnwMds it means it timed out the first request to the server after ~1s, and then refuses to use it again.

**Comment by csagan5** [comment]
@SnwMds you should get those logs while the issue is happening. At some point it uses a much lower timeout:
```
I chromium: [INFO:dns_session.cc(331)] DoH: using timeout: 1.01 s
I chromium: [INFO:dns_session.cc(331)] DoH: using timeout: 0.157 s
```

I will check if this can be fixed by changing the minimum timeout on next release.

**Comment by csagan5** [comment]
Fixed in `79.0.3945.111`.

**Comment by csagan5** [comment]
This issue seems to be not resolved yet.

**Comment by SnwMds** [comment]
> This issue seems to be not resolved yet.

Yes, it seems so. I did some tests with the version `79.0.3945.111` and noticed that the frequency with which requests times out has decreased, but the bug can still be reproduced in some cases.

**Comment by csagan5** [comment]
In next release I will enforce a minimum timeout of 400ms for DoH DNS requests, that should mitigate the issue.

**Comment by csagan5** [comment]
Fixed in `79.0.3945.123`.
